html/themes/custom/common_design_subtheme/README.md

```diff
@@ -53,26 +72,64 @@ Add new components by [defining new libraries](https://www.drupal.org/docs/themi
+
+  [library-override]: https://www.drupal.org/docs/theming-drupal/adding-stylesheets-css-and-javascript-js-to-a-drupal-theme#override-extend
 
 
 ### Extending Drupal libraries
 
-Core and CD base-theme libraries [can be extended with small customizations](https://www.drupal.org/docs/theming-drupal/adding-stylesheets-css-and-javascript-js-to-a-drupal-theme#s-libraries-extend) by using the `libraries-extend` directive in `common_design_subtheme.info.yml`. By extending a library, your customizations automatically apply when any core or base-theme library would normally be output. Use this feature to override colors inside components, for example.
+Core and CD base-theme libraries [can be extended with small customizations][library-extend] by using the `libraries-extend` directive in `common_design_subtheme.info.yml`. By extending a library, your customizations automatically apply when any core or base-theme library would normally be output. Use this feature to override colors inside components, for example.
+
+  [library-extend]: https://www.drupal.org/docs/theming-drupal/adding-stylesheets-css-and-javascript-js-to-a-drupal-theme#s-libraries-extend
 
 
 ### Extending the theme
 
-Override theme preprocess functions by copying from `common_design.theme` and editing as needed. For example, if new icons are added, a new icon sprite will need to be generated and the `common_design_preprocess_html` hook used to attach
-the icon sprite to the page will need a new path to reflect the sub theme's icon sprite location.
+Override theme preprocess functions by copying from `common_design.theme` and editing as needed. hen copying, the **function name will always need to be modified** from `common_design_` to `common_design_subtheme_`.
 
-Refer to [common_design README](https://github.com/UN-OCHA/common_design/#common-design-base-theme-for-drupal-89) for
+Refer to [common_design README][cd-readme] for
 general details about base-theme and instructions for compilation. There should be no need to use Sass in the sub-theme anymore.
 
+  [cd-readme]: https://github.com/UN-OCHA/common_design/blob/main/README.md#ocha-common-design-for-drupal-9
+
+
+### Tests
+
+To run E2E tests in the sub theme, adjust the Base URLs in the
+`common_design_subtheme/_tests/_env/index.js` file.
+
+```sh
+# Install dependencies for your host machine.
+npm i
+
+# Run all E2E tests in headless mode. The console will output the results.
+npm run e2e
 
-## Tests
+# See the tests run in a visible browser window with --debug
+npm run e2e -- --debug
 
-Refer to [common_design README E2E testing](https://github.com/UN-OCHA/common_design/#e2e-testing) for information about running tests.
+# If you want to run a limited number of tests, specify a string with the -t
+# argument. It will parse all of the describe() blocks and only run tests when
+# it matches the string you supply.
+
+# All tests that include 'OCHAServicesDropdown'.
+npm run e2e -- -t 'OCHAServicesDropdown'
+
+```
+
+### PHP CS
```

Comment:
This was not needed for UNOCHA, and it should probably not be in the base/sub theme readme.
See comments on https://github.com/UN-OCHA/common_design/pull/431

I think the Github actions tests in the repo https://github.com/UN-OCHA/common-design-site/blob/develop/.github/workflows/run-tests.yml might be better being updated to match https://github.com/UN-OCHA/drupal-starterkit/blob/develop/.github/workflows/run-tests.yml and then we likely won't need this exclusion rule.

Reply:
CD-506